core/src/test/resources/org/apache/calcite/test/RelOptRulesTest.xml

```diff
@@ -9173,6 +9266,131 @@ LogicalAggregate(group=[{0, 1}])
       <![CDATA[SELECT mgr, comm FROM emp
 EXCEPT
 SELECT mgr, comm FROM emp WHERE comm = 5
+]]>
+    </Resource>
+  </TestCase>
+  <TestCase name="testMinusToFilterRule3">
+    <Resource name="sql">
+      <![CDATA[SELECT deptno FROM emp WHERE deptno > 6
+EXCEPT
+SELECT deptno FROM dept WHERE deptno > 8
+EXCEPT
+SELECT deptno FROM emp WHERE deptno > 12
+EXCEPT
+SELECT deptno FROM dept WHERE deptno > 10
+]]>
+    </Resource>
+    <Resource name="planBefore">
+      <![CDATA[
+LogicalMinus(all=[false])
+  LogicalFilter(condition=[>($0, 6)])
+    LogicalProject(DEPTNO=[$7])
+      LogicalTableScan(table=[[CATALOG, SALES, EMP]])
+  LogicalFilter(condition=[>($0, 8)])
+    LogicalProject(DEPTNO=[$0])
+      LogicalTableScan(table=[[CATALOG, SALES, DEPT]])
+  LogicalFilter(condition=[>($0, 12)])
+    LogicalProject(DEPTNO=[$7])
+      LogicalTableScan(table=[[CATALOG, SALES, EMP]])
+  LogicalFilter(condition=[>($0, 10)])
+    LogicalProject(DEPTNO=[$0])
+      LogicalTableScan(table=[[CATALOG, SALES, DEPT]])
+]]>
+    </Resource>
+    <Resource name="planAfter">
+      <![CDATA[
+LogicalMinus(all=[false])
+  LogicalFilter(condition=[SEARCH($0, Sarg[(6..12]])])
+    LogicalProject(DEPTNO=[$7])
+      LogicalTableScan(table=[[CATALOG, SALES, EMP]])
+  LogicalFilter(condition=[>($0, 8)])
```

Comment:
`condition=[>($0, 8)]` this condition is right?

Reply:
Original SQL:
```
SELECT deptno FROM emp WHERE deptno > 6
EXCEPT
SELECT deptno FROM dept WHERE deptno > 8
EXCEPT
SELECT deptno FROM emp WHERE deptno > 12
EXCEPT
SELECT deptno FROM dept WHERE deptno > 10
```
Firstly can be convert to:
```
SELECT deptno FROM emp WHERE deptno > 6 and deptno <= 12
EXCEPT
(
SELECT deptno FROM dept WHERE deptno > 8
UNION
SELECT deptno FROM dept WHERE deptno > 10
)
```
Secondly can be convert to:
```
SELECT deptno FROM emp WHERE deptno > 6 and deptno <= 12
EXCEPT
SELECT deptno FROM dept WHERE deptno > 8
```
So it should be consistent with the current plan.